docs(docker): add ghworkflows example for GAR with Workload Identity (#30692)
renovatebot/renovate: docs/usage/docker.md

#### Google Container Registry / Google Artifact Registry

##### Using Workload Identity

> [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) has to be configured and the Service Account needs `artifactregistry.repositories.downloadArtifacts` permission.

Reviewer: ```suggestion
[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) must be configured and the Service Account must have the `artifactregistry.repositories.downloadArtifacts` permission.
```

###### With Application Default Credentials (Self-Hosted only)

> Just configure [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc) as normal and _don't_ provide a username, password or token.

Reviewer: ```suggestion
Configure [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc) as normal.
Do not provide a username, password or token.
```

> Renovate will automatically retrieve the credentials using the google-auth-library.

Reviewer: ```suggestion
Renovate will get the right credentials with the [`google-auth-library`](https://www.npmjs.com/package/google-auth-library).
```

Please check if I'm linking to the correct package!

###### With short-lived access token / GitHub Actions (Self-hosted only)

> Below you find the example configuration for both Workload Identity and the Renovate Host rules. For a full GitHub Workflow example see [renovatebot/github-action](https://github.com/renovatebot/github-action) repo.

Reviewer: ```suggestion
Example configuration for the Workload Identity and Renovate's host rules:
```

```yaml
- name: authenticate to google cloud
  id: auth
  uses: google-github-actions/auth@v2.1.3
  with:
    token_format: 'access_token'
    workload_identity_provider: ${{ env.WORKLOAD_IDENTITY_PROVIDER }}
    service_account: ${{ env.SERVICE_ACCOUNT }}

- name: renovate
  uses: renovatebot/github-action@v40.2.4
  env:
    RENOVATE_HOST_RULES: |
      [
        {
          matchHost: "us-central1-docker.pkg.dev",
          hostType: "docker",
          username: "oauth2accesstoken",
          password: "${{ steps.auth.outputs.access_token }}"
        }
      ]
  with:
    token: ${{ secrets.RENOVATE_TOKEN }}
    configurationFile: .github/renovate.json5
```

> 

Reviewer: ```suggestion

You can find a [full GitHub Workflow example on the `renovatebot/github-action` repository](https://github.com/renovatebot/github-action).
```

#### Google Container Registry / Google Artifact Registry

##### Using Workload Identity

[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) must be configured and the Service Account must have the `artifactregistry.repositories.downloadArtifacts` permission.

###### With Application Default Credentials (Self-Hosted only)

> Configure [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc) as normal.
> Do not provide a username, password or token.
> Renovate will get the credentials with the [`google-auth-library`](https://www.npmjs.com/package/google-auth-library).

Reviewer: ```suggestion
To let Renovate authenticate with [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc), you must:

- Configure ADC as normal
- _Not_ provide a username, password or token

Renovate will get the credentials with the [`google-auth-library`](https://www.npmjs.com/package/google-auth-library).
```

> ###### With short-lived access token / GitHub Actions (Self-hosted only)

Reviewer: ```suggestion
###### With short-lived access token / GitHub Actions (self-hosted only)
```

Example configuration for Workload Identity and Renovate's host rules:

> ```yaml

Reviewer: Please:

1. Delete line 294: `Example configuration for Workload Identity and Renovate's host rules:`.
2. Add a title to the YAML codeblock. [^1]

The bit after the three backticks should look like this:

`yaml title="Example for Workload Identity plus Renovate host rules"`

[^1]: [Material for MkDocs, adding a title to a codeblock](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title)

      }
      ```

> ##### Using short-lived access token / Gitlab CI / Google Cloud

Reviewer: This section feels like a wall of text. Can you add some headings to break it up into smaller parts?

Author: out of scope for this pr

Reviewer: I'll make a follow up PR later then.
